pkg/cmd/config.go

```diff
@@ -0,0 +1,93 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/KubeHound/pkg/config"
+	"github.com/DataDog/KubeHound/pkg/telemetry"
+	"github.com/DataDog/KubeHound/pkg/telemetry/log"
+	"github.com/DataDog/KubeHound/pkg/telemetry/tag"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+func GetConfig() (*config.KubehoundConfig, error) {
+	// Passing the Kubehound config from viper
+	khCfg := config.KubehoundConfig{}
+	err := viper.Unmarshal(&khCfg)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal viper: %w", err)
+	}
+
+	return &khCfg, nil
+}
+
+func InitializeKubehoundConfig(ctx context.Context, configPath string, generateRunID bool) error {
+	// We define a unique run id this so we can measure run by run in addition of version per version.
+	// Useful when rerunning the same binary (same version) on different dataset or with different databases...
+	// In the case of KHaaS, the runID is taken from the GRPC request argument
+	if generateRunID {
+		viper.Set(config.DynamicRunID, config.NewRunID())
+	}
+
+	clusterName, err := config.GetClusterName(ctx)
+	if err == nil {
+		viper.Set(config.DynamicClusterName, clusterName)
+	} else {
+		log.I.Errorf("collector cluster info: %v", err)
+	}
```

Comment:
shouldn't we _fail_ if we can't get the cluster?
Or is this to support runs without a cluster (from a dump?) ? If so, that's worth a comment on that edge case (It's not clear to me _why_ we would want to continue when something this fundamental isn't present.

Also, I don't believe there's a check for `clusterName = ""` which I believe could be frequent since it's a the default value of a string?

Reply:
This code is also used for file ingestion (dump), so it is not needed in this case. So, we can continue if it fails.